### PR TITLE
nuttx net warning:add initialization of pointer *aiocbp in lio_listio

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
and nuttx net warning:cancel the conflict between weak and inline in unwind_arch_adjustment

## Impact
Warning fix

### CODE REVIEW

File: libs/libc/aio/lio_listio.c [hunk 1]
```diff
@@ -507,7 +507,7 @@ static int lio_waitall(FAR struct aiocb * const *list, int nent)
 int lio_listio(int mode, FAR struct aiocb * const list[], int nent,
                FAR struct sigevent *sig)
 {
-  FAR struct aiocb *aiocbp;
+  FAR struct aiocb *aiocbp = NULL;
```
Comment (gustavonihei) [inline libs/libc/aio/lio_listio.c:510]: According to the NuttX coding standard, `aiocbp` should be initialized separately from its definition.
So it is better to move the NULL attribution to line 523

Comment (xiaoxiang781216) [reply libs/libc/aio/lio_listio.c:510]: Could you point out the rule? There are many many places initialize the variables directly.

Comment (gustavonihei) [reply libs/libc/aio/lio_listio.c:510]: Sorry, that was a misinterpretation of the C89 standard from myself. LGTM

File: libs/libc/debug/lib_backtrace.c [hunk 2]
```diff
@@ -44,7 +44,7 @@ struct trace_arg
  * Private Functions
  ****************************************************************************/
 
-static inline weak_function
+weak_function
```
Comment (gustavonihei) [inline libs/libc/debug/lib_backtrace.c:47]: Could you at least move this change to a separate commit? This is not related to the above change.

Comment (xiaoxiang781216) [reply libs/libc/debug/lib_backtrace.c:47]: Done, thanks.

**Review by gustavonihei — APPROVED** [review]
